Makefile

```diff
@@ -18,3 +20,6 @@ fix-format:
 
 check-spell:
 	@.github/codespell.sh
+
+tests: build
+	src/tools/softhsm/test.sh
```

Comment:
should we run it with the `check` targets we have instead of reinventing a new one?

Reply:
Or ad this to the CI so we can see it works in different configurations.

Reply:
I have another commit I forgot for the CI, we can't really run as part of check because check does a cargo test, and we need a cargo build, and I do not want to do both for the common check case.